tests/conda_local_channel/provision_wheels.py

```diff
@@ -0,0 +1,187 @@
+"""
+Provision wheel files for the conda local channel test fixture.
+
+This script downloads the wheel files referenced in repodata.json
+from PyPI and verifies their checksums.
+"""
+
+import hashlib
+import json
+import sys
+from pathlib import Path
+from urllib import request
+from urllib.error import HTTPError, URLError
+
+
+HERE = Path(__file__).parent
+NOARCH = HERE / "noarch"
+REPODATA = NOARCH / "repodata.json"
+
+
+def checksum(path: Path, algorithm: str = "sha256") -> str:
+    """Calculate checksum of a file."""
+    hasher = hashlib.new(algorithm)
+    with open(path, "rb") as f:
+        for chunk in iter(lambda: f.read(1024 * 1024), b""):
```

Comment:
nice usage of the second sentinel argument to `iter`. For "quick and cheerful" a big-buffer `hashlib.new(algorithm)(file.read_bytes())` also works. In conda-index we chose a `1<<18=262144` buffer after experimentation.